Xamarin.Forms.Platform.Android/FastRenderers/FrameRenderer.cs

```diff
@@ -80,9 +81,12 @@ void IVisualElementRenderer.SetElement(VisualElement element)
 			var frame = element as Frame;
 			if (frame == null)
 				throw new ArgumentException("Element must be of type Frame");
+			VisualElement oldElement = Element;
 			Element = frame;
 			_motionEventHelper.UpdateElement(frame);
 
+			EffectUtilities.RegisterEffectControlProvider(this, oldElement, element);
```

Comment:
Per @hartez 
`FastRenderers.VisualElementRenderer` is a convenience class which bundles up a bunch of common stuff that many renderers need (an AutomationPropertiesProvider, GestureManager, EffectControlProvider, TabStop ...). All of these services _can_ be added on to a renderer individually, but having them in one bundle is convenient for renderers which need them all.
At the moment, it's used by the fast LabelRenderer and the fast ImageRenderer. It's _not_ used by ButtonRenderer because Buttons _don't_ need all of those services; notably, Buttons don't have gestures.

Had we implemented other fast renderers, several of them probably would have used `FastRenderers.VisualElementRenderer`.

FrameRenderer doesn't use `FastRenderers.VisualElementRenderer` because it pre-dates all of the other fast renderers (in fact, it's the _original_ fast renderer; the AppCompat FrameRenderer _never_ used the "wrapped in a VisualElementRenderer<T>" design that the other controls did). TBH, you probably _could_ update FrameRenderer to use `FastRenderers.VisualElementRenderer` at this point; I _think_ it uses all of the services. But I'm not sure it's worth the effort.

> I'm inclined to let the logic through per the Button convention

Probably the easiest and safest option.